src/hotspot/share/utilities/ostream.cpp

```diff
@@ -360,21 +380,21 @@ void stringStream::write(const char* s, size_t len) {
 }
 
 void stringStream::zero_terminate() {
-  assert(buffer != NULL &&
-         buffer_pos < buffer_length, "sanity");
-  buffer[buffer_pos] = '\0';
+  assert(_buffer != NULL &&
+         _written < _capacity, "sanity");
+  _buffer[_written] = '\0';
 }
 
 void stringStream::reset() {
-  buffer_pos = 0; _precount = 0; _position = 0;
+  _written = 0; _precount = 0; _position = 0;
```

Comment:
Strictly speaking yes. But _newlines is really only used as a very circumvent way to trigger a flush in defaultStream::write() if the last write contained a newline. We could probably throw _newlines completely away at the cost of a `strchr(s, '\n')` in defaultStream::write(). I'll remove the outside accessor `newlines()` at least, since its not used, and will correct stringStream::reset(), so it is consistent for now. I leave the cleanup for another day.